crates/net/network/src/transactions.rs

```diff
@@ -300,6 +309,10 @@ where
         to_propagate: Vec<PropagateTransaction>,
     ) -> PropagatedTransactions {
         let mut propagated = PropagatedTransactions::default();
+        #[cfg(feature = "optimism")]
+        if self.network.tx_gossip_disabled() {
+            return propagated
+        }
```

Comment:
I'm okay with making this feature non cfg optional, just makes it easier

Reply:
Made optional, but kept sequencer endpoint within an outlined optimism struct in https://github.com/anton-rs/op-reth/pull/141